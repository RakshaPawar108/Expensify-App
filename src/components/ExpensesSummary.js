import React from "react";
import selectExpenses, { getAllExpenses } from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import numeral from "numeral";

export const ExpensesSummary = (props) => (
  <div className="page-header">
    <div className="content-container">
      <h1 className="page-header__title">
        Viewing <span>{props.expenseCount}</span>{" "}
        {props.expenseCount === 1 ? "expense" : "expenses"} totalling{" "}
        <span>{props.expensesTotal}</span>
      </h1>
      {props.hiddenExpenses > 0 ? (
        <p className="page-header__subtitle">
          There {props.hiddenExpenses > 1 ? "are" : "is"}{" "}
          <span>{props.hiddenExpenses}</span> expense
          {props.hiddenExpenses > 1 ? "s" : ""} hidden.
        </p>
      ) : null}
      <div className="page-header__actions">
        <Link className="button" to="/create">
          Add Expense
        </Link>
      </div>
    </div>
  </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: numeral(selectExpensesTotal(visibleExpenses) / 100).format(
      "$0,0.00"
    ),
    hiddenExpenses: getAllExpenses(state.expenses, visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
