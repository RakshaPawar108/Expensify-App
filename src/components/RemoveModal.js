import React from "react";
import Modal from "react-modal";

const RemoveModal = (props) => (
  <Modal
    isOpen={!!props.promptRemove}
    onRequestClose={props.handleClearPromptRemove}
    contentLabel="Are you sure you want to remove this expense?"
    closeTimeoutMS={200}
    className="modal"
    appElement={app}
  >
    <p className="modal__title">
      Are you sure you want to remove this expense?
    </p>
    <div className="input-group modal-input">
      <button
        className="button button--delete input-group__item"
        onClick={props.onRemoveConfirm}
      >
        Remove
      </button>
      <button
        className="button button--secondary input-group__item"
        onClick={props.handleClearPromptRemove}
      >
        Cancel
      </button>
    </div>
  </Modal>
);

export default RemoveModal;
