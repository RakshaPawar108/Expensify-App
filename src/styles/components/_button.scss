.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}

.button:focus {
  outline: none;
}

.button:active {
  transform: scale(0.9);
}
