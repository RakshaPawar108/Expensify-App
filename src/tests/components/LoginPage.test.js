import React from "react";
import { LoginPage } from "../../components/LoginPage";
import { shallow } from "enzyme";

test("Should render login page correctly", () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test("Should call startLogin on button click", () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find("button").simulate("click");
  expect(startLogin).toHaveBeenCalled();
});
