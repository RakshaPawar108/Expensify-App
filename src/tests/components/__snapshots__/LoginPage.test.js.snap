// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render login page correctly 1`] = `
<div>
  <h1>
    Login to Expensify
  </h1>
  <button>
    Login
  </button>
</div>
`;
